STM32/Filtering/Src/array-math.c

```diff
@@ -0,0 +1,106 @@
+/*! 
+**  \brief     Array math functions
+**  \details   Functions to find the max/average of arrays
+**  \author    Luke Walker
+**  \date      09/04/2024
+**/
+
+#include "array-math.h"
+
+/***************************************************************************************************
+** PUBLIC FUNCTION DEFINITIONS
+***************************************************************************************************/
+
+/*!
+** @brief Returns the maximum value in an array of double
+*/
+double max_element(double arr[], int len) {
+    if(len != 0) {
+        double ret_val = arr[0];
+        for(int i = 1; i < len; i++) {
+            if(arr[i] > ret_val) ret_val = arr[i];
+        }
+        return ret_val;
+    }
+
+    return -1.0;
```

Comment:
How do you distinguish between a return value of -1.0 meaning an error occured and the actual max element being -1.0?

You could add another input parameter which could be a double pointer that is assigned and the return value is a status value. 

Otherwise we could return -DBL_MAX which is the minimum value double value - but it seems you have to #include "float.h"

Reply:
Worked out a solution - it seems overly robust, but it does the job.